stablehlo/dialect/TypeInference.h

```diff
@@ -165,6 +210,12 @@ LogicalResult inferConstantOp(Optional<Location>, ElementsAttr value,
 LogicalResult inferCreateTokenOp(Dialect* dialect, Optional<Location> location,
                                  SmallVectorImpl<Type>& inferredReturnTypes);
 
+LogicalResult inferDynamicGatherOp(
+    Optional<Location> location, ValueShapeRange operands,
```

Comment:
Looks like ShapeAdaptor can be created from Type, so how about we pass strongly-typed values here? (I.e. `operand`, `startIndices` and `sliceSizes). Same for GatherOp.

Reply:
Nice suggestion! Changed.